skimage/future/detect_obj/cascade.pyx

```diff
@@ -0,0 +1,303 @@
+        MBLBPStump * stumps
+        MBLBP * features
+        cnp.uint32_t * LUTs
+
+    def __dealloc__(self):
+
+        # Free the memory that was used for c-arrays.
+        free(self.stages)
+        free(self.stumps)
+        free(self.features)
+        free(self.LUTs)
+
+    cdef int evaluate(self, float[:, ::1] int_img, Py_ssize_t row, Py_ssize_t col, float scale) nogil:
+
+        cdef:
+            float stage_threshold
+            float stage_points
+            int lbp_code
+            int bit
+            Py_ssize_t stage_number
+            Py_ssize_t weak_classifier_number
+            Py_ssize_t feature_number
+            Py_ssize_t features_amount
+            Py_ssize_t stumps_amount
+            Py_ssize_t first_stump_idx
+            Py_ssize_t lut_idx
+            Py_ssize_t r, c, widht, height
+            cnp.uint32_t[::1] current_lut
+            Stage current_stage
+            MBLBPStump current_stump
+            MBLBP current_feature
+
+
+        for stage_number in range(self.stages_amount):
+
+            current_stage = self.stages[stage_number]
+            first_stump_idx = current_stage.first_idx
+            stage_points = 0
+
+            for weak_classifier_number in range(current_stage.amount):
+
+                current_stump = self.stumps[first_stump_idx + weak_classifier_number]
+
+                current_feature = self.features[current_stump.feature_id]
+
+                r = <Py_ssize_t>(current_feature.r * scale)
+                c = <Py_ssize_t>(current_feature.c * scale)
+                width = <Py_ssize_t>(current_feature.width * scale)
+                height = <Py_ssize_t>(current_feature.height * scale)
+
+
+                lbp_code = _multiblock_lbp(int_img,
+                                           row + r,
+                                           col + c,
+                                           width,
+                                           height)
+
+                lut_idx = current_stump.lut_idx
+
+                bit = (self.LUTs[lut_idx + (lbp_code >> 5)] >> (lbp_code & 31)) & 1
+
+                stage_points += current_stump.left if bit else current_stump.right
+
+            if stage_points < (current_stage.threshold - self.eps):
+                return 0
+
+        return 1
+
+    def get_valid_scale_factors(self, min_size, max_size, scale):
+
+        min_size = np.array(min_size)
+        max_size = np.array(max_size)
+
+        scale_factors = []
+        current_scale = 1
+        current_size = np.array((self.window_height, self.window_width))
+
+        while (current_size <= max_size).all():
+
+            if (current_size >= min_size).all():
+                scale_factors.append(current_scale)
+
+            current_scale = current_scale * scale
+            current_size = current_size * scale
+
+        return scale_factors
+
+    def detect_single_scale(self, float[:, ::1] int_img, float scale, int step_ratio=1, int amount_of_threads=4):
```

Comment:
@vighneshbirodkar this occurs because when we spawn threads for each scale they have unequal amount of work (the first scales are the hardest ones) and, therefore, we need to allow them to take up more resources when they need it.

So, we stick with scale threading and `dynamic`. I hope that my tests were right :)